docs/developer/architecture/configurations.rst

```diff
@@ -73,3 +73,205 @@ As a MetalK8s expert, I can use ``kubectl`` command(s) in order to edit all
+  ConfigMaps and hence will be catered for later.
+
+- As stated in the requirements, adding and editing Prometheus alert rules
+  is also not covered by the chosen design choice and will need to be addressed
+  differently.
+
+Rejected design choices
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Consul KV vs ConfigMap
+~~~~~~~~~~~~~~~~~~~~~~
+
+This approach offers a full fledge KV store with a /kv endpoint which allows
+CRUD operations on all KV data stored in it.
+Consul KV also allows access to past versions of objects and has an optimistic
+concurrency when manipulating multiple objects.
+
+Note that, Consul KV store was rejected because managing operations such as
+performing full backups, system restores for a full fledged KV system
+requires time and much more efforts than the ConfigMap approach.
+
+Operator (Custom Controller) vs Salt
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Operators are useful in that, they provide self-healing functionalities on a
+reactive basis. When a user changes a given configuration, it is easy to
+reconcile and apply these changes to the in-cluster objects.
+
+The Operator approach was rejected because it is much more complex, requires
+much more effort to realize and there is no real need for applying changes
+using this method because configuration changes are not frequent
+(for a typical MetalK8s admin, changing the number of replicas for a given
+deployment could happen once in 3 months or less) as such, having an operator
+watch for object changes is not significant and not very useful at this point
+in time.
+
+
+Implementation Details
+----------------------
+
+Iteration 1
+^^^^^^^^^^^
+
+- Define and deploy new ConfigMap stores that will hold cluster and service
+  configurations as listed in the requirements
+- Template and render Deployment and Pod manifests that will make use of
+  this persisted cluster and service configurations
+- Document how to change cluster and service configurations using kubectl
+- Document the entire list of configurations which can be changed by the user
+
+Iteration 2
+^^^^^^^^^^^
+
+- Provide a CLI tool for changing any of the cluster and service
+  configurations:
+
+    - Count of replicas for chosen Deployments (Prometheus)
+    - Updating a Dex authentication connector (OpenLDAP, AD and
+      staticUser store)
+    - Updating the Alertmanager notification configuration
+
+- Provide a UI interface for adding, updating and deleting service specific
+  configurations for example Dex-LDAP connector integration.
+- Provide a UI interface for listing MetalK8s available/supported Dex
+  authentication Connectors
+- Provide a UI interface for enabling or disabling Dex authentication
+  connectors (LDAP, Active Directory, StaticUser store)
+- Provide a UI interface for changing the number of replicas for a chosen set
+  of MetalK8s deployments (Prometheus, ....)
+- Add a UI interface for listing Alertmanager notification systems MetalK8s
```

Comment:
It's not that "MetalK8s will support", we want to allow anything Alertmanager supports, IMO.

The comment from Nicolas was just that HipChat is EOL, so no need to mention it as an example (no one will use it).

Reply:
We are talking about 9 different Alertmanager channels for notifications and i am not sure we want to support all those IMHO.